lib/rules/no-manual-cleanup.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const { getDocsUrl } = require('../utils');
+
+const findCleanupSpecifier = specifiers => {
+  return specifiers.find(specifier => specifier.imported.name === 'cleanup');
+};
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: ' Disallow the use of `cleanup`',
+      category: 'Best Practices',
+      recommended: false,
+      url: getDocsUrl('no-manual-cleanup'),
+    },
+    messages: {
+      noManualCleanup:
+        "`cleanup` is performed automatically after each test by {{library}}, you don't need manual cleanups.",
```

Comment:
This is not right: the automatic cleanup is performed depending on your testing framework, not on your testing library package. This rule should just mention that manual cleanups are dissalowed as the testing framework taking care of it automatically.

Reply:
We are both not right. While it's true the automatic cleanup is performed depending on the testing framework, not all testing libraries package have a `cleanup` method, like Angular's testing library. 

It does not make sense for me to make this rule fire even if the package doesn't have a `cleanup` method.

I agree this rule should be removed from shareable configs as we don't know the test runner used though.

Reply:
Fair enough about short-circuiting when importing from a testing library package without `cleanup`, but still the error message is wrong as the one handling the automatic cleanup is the testing framework (e.g. jest), not testing library package (e.g. `@testing-library/angular`).

(Damn, "testing library" and "testing framework" are not making easier this discussion).

Reply:
Sure, I'll fix it too, thanks for the feedback! 👍

By the way, there are some rules like `no-debug` where we don't support frameworks like Svelte Testing Library, we should probably open a new issue to include them.

Reply:
That's actually one of the reasons I don't like to write code for checking specific testing library packages as if a new one comes up tomorrow we could have to update rules. I prefer to leave them agnostic from any package, specially those not enabled by default, so users can set them up under their responsibility. Of course, we should mention in proper rule doc which are those testing library packages supposed to be compatible with the rule for clarification.

Reply:
What I meant is that we have shareable configs for Angular, Vue, etc. But we are not adding any support for Svelte, Preact, etc. I think we should extend the rules that already work with specific packages (such as no-debug, or no-dom-import) to other potential testing libraries 🙂

Reply:
Ah of course 🤦‍♂️

And probably we should open an issue about including more frameworks for sure!